packages/loot-core/migrations/1723665565000_prefs.js

```diff
@@ -0,0 +1,68 @@
+const SYNCED_PREF_KEYS = [
+  'firstDayOfWeekIdx',
+  'dateFormat',
+  'numberFormat',
+  'hideFraction',
+  'isPrivacyEnabled',
+  /^show-extra-balances-/,
+  /^hide-cleared-/,
+  /^parse-date-/,
+  /^csv-mappings-/,
+  /^csv-delimiter-/,
+  /^csv-has-header-/,
+  /^ofx-fallback-missing-payee-/,
+  /^flip-amount-/,
+  // 'budgetType', // TODO: uncomment when `budgetType` moves from metadata to synced prefs
+  /^flags\./,
+];
+
+export default async function runMigration(db, { fs, fileId }) {
+  await db.execQuery(`
+    CREATE TABLE preferences
+       (id TEXT PRIMARY KEY,
+        value TEXT);
+  `);
+
+  try {
+    const budgetDir = fs.getBudgetDir(fileId);
+    const fullpath = fs.join(budgetDir, 'metadata.json');
+
+    const prefs = JSON.parse(await fs.readFile(fullpath));
+
+    if (typeof prefs !== 'object') {
+      return;
+    }
+
+    await Promise.all(
+      Object.keys(prefs).map(async key => {
+        // Check if the current key is of synced-keys type
+        if (
+          !SYNCED_PREF_KEYS.find(keyMatcher =>
+            keyMatcher instanceof RegExp
+              ? keyMatcher.test(key)
+              : keyMatcher === key,
+          )
+        ) {
+          return;
+        }
+
+        // insert the synced prefs in the new table
+        await db.runQuery('INSERT INTO preferences SET id = ?, value = ?', [
+          key,
+          String(prefs[key]),
+        ]);
+
+        // remove the synced prefs from the metadata file
+        delete prefs[key];
+      }),
+    );
+
+    // Update the metadata.json file
+    await fs.writeFile(
```

Comment:
This seems dangerous - is there any harm in keeping the old metadata around in case the user needs to rollback to a prev version of the client?

Reply:
Good shout! I've removed the `metadata.json` updates: https://github.com/actualbudget/actual/pull/3423/commits/397d5993cbad41526bddb11486aeaf5947333075